core/src/main/java/org/bitcoinj/core/listeners/DownloadProgressTracker.java

```diff
@@ -89,11 +90,23 @@ public void onBlocksDownloaded(Peer peer, Block block, @Nullable FilteredBlock f
      * Called when download progress is made.
      *
      * @param pct  the percentage of chain downloaded, estimated
-     * @param date the date of the last block downloaded
+     * @param time the time of the last block downloaded
      */
-    protected void progress(double pct, int blocksSoFar, Date date) {
+    protected void progress(double pct, int blocksSoFar, Instant time) {
         log.info(String.format(Locale.US, "Chain download %d%% done with %d blocks to go, block date %s", (int) pct, blocksSoFar,
-                TimeUtils.dateTimeFormat(date)));
+                TimeUtils.dateTimeFormat(time.toEpochMilli())));
+        progress(pct, blocksSoFar, new Date(time.toEpochMilli()));
+    }
+
+    /**
+     * Called when download progress is made.
+     *
+     * @param pct  the percentage of chain downloaded, estimated
+     * @param time the time of the last block downloaded
+     * @deprecated use {@link #progress(double, int, Instant)}
+     */
+    @Deprecated
+    protected void progress(double pct, int blocksSoFar, Date time) {
```

Comment:
Should this do one of the following:
* Call the new method?
* Throw a RuntimeException?
* Be removed entirely to cause a breaking compile error?

Reply:
Well, that's the question. I had the idea that a user either overrides one or the other.

It should be noted that this method is not meant to be invoked by users, it's more like a callback.

Reply:
Of course, we could simply switch the signature in those cases, and not provide backwards compatibility. Is that what you prefer?

Reply:
I think that's reasonable and probably simpler for everyone.

Reply:
Ok, I removed the deprecated method and its invocation.